integration-cli/docker_cli_start_test.go

```diff
@@ -186,6 +186,7 @@ func (s *DockerSuite) TestStartAttachWithRename(c *testing.T) {
 }
 
 func (s *DockerSuite) TestStartReturnCorrectExitCode(c *testing.T) {
+	skip.If(c, DaemonIsWindows(), "FLAKY on Windows, see #38521")
```

Comment:
I see `skip` is not imported here, so this will probably fail. If it's only flaky on RS1, we should skip it on `< RS3` (or `< RS5`), so that we still run the test on RS5;

```suggestion
	// Note we parse kernel.GetKernelVersion rather than system.GetOSVersion
	// as test binaries aren't manifested, so would otherwise report the wrong
	// build number.
	if runtime.GOOS == "windows" {
		v, err := kernel.GetKernelVersion()
		assert.NilError(c, err)
		build, _ := strconv.Atoi(strings.Split(strings.SplitN(v.String(), " ", 3)[2][1:], ".")[0])
		if build < 16299 {
			c.Skip("FLAKY on Windows RS1, see #38521")
		}
	}
```

Patch below:


<details>

```patch
diff --git a/integration-cli/docker_cli_start_test.go b/integration-cli/docker_cli_start_test.go
index 398ef242cf..72a5d77d34 100644
--- a/integration-cli/docker_cli_start_test.go
+++ b/integration-cli/docker_cli_start_test.go
@@ -2,11 +2,14 @@ package main
 
 import (
        "fmt"
+       "runtime"
+       "strconv"
        "strings"
        "testing"
        "time"
 
        "github.com/docker/docker/integration-cli/cli"
+       "github.com/docker/docker/pkg/parsers/kernel"
        "gotest.tools/assert"
        "gotest.tools/icmd"
 )
@@ -186,7 +189,18 @@ func (s *DockerSuite) TestStartAttachWithRename(c *testing.T) {
 }
 
 func (s *DockerSuite) TestStartReturnCorrectExitCode(c *testing.T) {
-       skip.If(c, DaemonIsWindows(), "FLAKY on Windows, see #38521")
+       // Note we parse kernel.GetKernelVersion rather than system.GetOSVersion
+       // as test binaries aren't manifested, so would otherwise report the wrong
+       // build number.
+       if runtime.GOOS == "windows" {
+               v, err := kernel.GetKernelVersion()
+               assert.NilError(c, err)
+               build, _ := strconv.Atoi(strings.Split(strings.SplitN(v.String(), " ", 3)[2][1:], ".")[0])
+               if build < 16299 {
+                       c.Skip("FLAKY on Windows RS1, see #38521")
+               }
+       }
+
        dockerCmd(c, "create", "--restart=on-failure:2", "--name", "withRestart", "busybox", "sh", "-c", "exit 11")
        dockerCmd(c, "create", "--rm", "--name", "withRm", "busybox", "sh", "-c", "exit 12")
```

</details>

Reply:
i've updated with your proposed patch